### PR TITLE
doc: Update Requirements section in Bluetooth samples

### PR DESCRIPTION
**Pull request by umapraseeda**
ref: NCSDK-4950
Update the Requirements section in Bluetooth samples

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

### CODE REVIEW

**Review by trond-snekvik — COMMENTED** [review]
Are all these new platforms added to the sample.yaml files, so we build them in CI? For instance, nRF52820 only has 32kB RAM, which could be too small for samples that have previously only been supported on nRF52840 and nRF52832.

Could `:sample-yaml-rows:` from https://github.com/nrfconnect/sdk-nrf/pull/2944 replace some of these, so we don't have to maintain two lists and risk incompatibilities?

**Review by trond-snekvik — APPROVED** [review]
Discussed rows-from-sample offline, agreed not to do this for this PR, as sample.yaml lists unsupported boards.

Approving for Bluetooth Mesh, but I think we should have a discussion on whether unsupported boards should be listed in sample.yaml. We're either supporting them or we're not, and it'd be nice to have a single source of truth for this, so we don't have to make PRs like this for both docs and sample.yaml in the future.

**Review by kapi-no — APPROVED** [review]

File: samples/bluetooth/direct_test_mode/README.rst [hunk 1]
```diff
@@ -164,12 +164,14 @@ If you want to view the debug messages, follow the procedure in :ref:`testing_rt
...
-* The following development kit:
+The sample supports the following development kit:
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
    :rows: nrf5340pdk_nrf5340_cpunet
 
+Additionally, the sample has the following requirements:
+
 * The Network core
```
Comment (ru-fu) [inline samples/bluetooth/direct_test_mode/README.rst:175]: Unrelated to this PR, but if any of the reviewers can explain what we mean when we require "the network core", this would be good to clarify. ;)
Maybe that the network core must be programmed with the empty firmware sample?

File: samples/bluetooth/enocean/README.rst [hunk 2]
```diff
@@ -22,13 +22,13 @@ The LEDs of the board also respond to button presses from an EnOcean switch.
...
 
-* One of the following development boards:
+The sample supports the following development kits:
 
-  * |nRF5340DK|
-  * |nRF52840DK|
-  * |nRF52DK|
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52840dk_nrf52811, nrf52833dk_nrf52833, nrf52833dk_nrf52820, nrf52dk_nrf52810
```
Comment (ru-fu) [inline samples/bluetooth/enocean/README.rst:29]: Maybe change the order to keep the two nRF52840 and the two nRF52 entries together?

File: samples/bluetooth/llpm/README.rst [hunk 3]
```diff
@@ -79,15 +79,16 @@ By default, the following values are used to demonstrates the interaction of the
...
-  * |nRF52DK|
-  * |nRF5340DK|
-  * Other boards running SoftDevice Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support)
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns,nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf52dk_nrf52832
 
-  You can mix different boards.
-* Connection to a computer with a serial terminal for each of the boards.
+The sample also supports other development kits running BLE Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support).
```
Comment (ru-fu) [inline samples/bluetooth/llpm/README.rst:88]:
```suggestion
The sample also supports other development kits running Bluetooth LE Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support).
```

File: samples/bluetooth/peripheral_uart/README.rst [hunk 4]
```diff
@@ -34,15 +34,16 @@ If you want to view the debug messages, follow the procedure in :ref:`testing_rt
...
-  * |nRF52840DK|
-  * |nRF52DK|
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns, nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf52833dk_nrf52820, nrf52dk_nrf52832
 
-* A phone or tablet running a compatible application. The `Testing`_ instructions refer to nRF Connect for Mobile, but similar applications (for example, nRF Toolbox) can be used as well.
 
-  You can also test the application with the :ref:`central_uart` sample. See the documentation for that sample for detailed instructions.
+The sample also requires a phone or tablet running a compatible application. The `Testing`_ instructions refer to nRF Connect for Mobile, but similar applications (for example, nRF Toolbox) can be used as well.
```
Comment (ru-fu) [inline samples/bluetooth/peripheral_uart/README.rst:44]:
```suggestion
The sample also requires a phone or tablet running a compatible application. 
The `Testing`_ instructions refer to nRF Connect for Mobile, but similar applications (for example, nRF Toolbox) can be used as well.
```

File: samples/bluetooth/peripheral_uart/README.rst [hunk 5]
```diff
@@ -34,15 +34,16 @@ If you want to view the debug messages, follow the procedure in :ref:`testing_rt
...
+
+You can also test the application with the :ref:`central_uart` sample. See the documentation for that sample for detailed instructions.
```
Comment (ru-fu) [inline samples/bluetooth/peripheral_uart/README.rst:46]:
```suggestion
You can also test the application with the :ref:`central_uart` sample. 
See the documentation for that sample for detailed instructions.
```

**Comment by carlescufi** [comment]
@umapraseeda could you take a look at the review comments please?

File: samples/bluetooth/direct_test_mode/README.rst (hunk 1 above)
Comment (kapi-no) [reply samples/bluetooth/direct_test_mode/README.rst:175]: Its the other way around. This sample is meant to be run on the Network Core with empty firmware on the Application Core. However, a comment, mentioning this, is available in the build section of this file.

It is safe to remove this line. Previously, when listing sample compatible targets, we used DK names with links to UGs. I have probably added the Network Core part here to be more precise which core to use with nRF5340 DK. I think it is leftover from past changes.

**Comment by umapraseeda** [comment]
Thanks for the suggestions! Made the changes as suggested.

**Review by ru-fu — APPROVED** [review]
Looks good to me!

File: samples/bluetooth/central_dfu_smp/README.rst [hunk 6]
```diff
@@ -26,15 +26,13 @@ The response is decoded and displayed using the `TinyCBOR`_ library (which is pa
...
-  * |nRF5340DK|
-  * |nRF52840DK|
-  * |nRF52DK|
-
-* A device running `MCUmgr`_ with `SMP over Bluetooth`_, for example, another board running the :ref:`smp_svr_sample`
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns , nrf52840dk_nrf52840, nrf52840dk_nrf52811, nrf52833dk_nrf52833, nrf52833dk_nrf52820, nrf52dk_nrf52832, nrf52dk_nrf52810
 
+The sample also requires a device running `MCUmgr`_ with `SMP over Bluetooth`_, for example, another board running the :ref:`smp_svr_sample`.
```
Comment (joerchan) [inline samples/bluetooth/central_dfu_smp/README.rst:35]: I would prefer SMP_SVR to avoid confusion as to what "SMP" really means.

File: samples/bluetooth/peripheral_nfc_pairing/README.rst [hunk 7]
```diff
@@ -66,13 +66,15 @@ You can change the default role by choosing ``CONFIG_NFC_TAG_CH_SELECTOR`` or ``
...
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
    :rows: nrf5340pdk_nrf5340_cpuapp, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52833dk_nrf52833
 
-* NFC polling device (for example, a smartphone or a tablet with NFC support)
+The sample has the following additional requirements:
+
+* NFC polling device (for example, a smartphone or a tablet with NFC support).
 * NFC poling device with TNEP support (when testing the two-way OOB pairing).
```
Comment (lemrey) [inline samples/bluetooth/peripheral_nfc_pairing/README.rst:78]: typo `poling`/`polling`

File: samples/bluetooth/llpm/README.rst [hunk 8]
```diff
@@ -79,15 +79,16 @@ By default, the following values are used to demonstrates the interaction of the
...
-  * |nRF52DK|
-  * |nRF5340DK|
-  * Other boards running SoftDevice Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support)
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns,nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf52dk_nrf52832
 
-  You can mix different boards.
-* Connection to a computer with a serial terminal for each of the boards.
+The sample also supports other development kits running Bluetooth LE Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support).
```
Comment (joerchan) [inline samples/bluetooth/llpm/README.rst:88]: Why the change to Bluetooth LE controller instead of SoftDevice controller?
Variants refers to variants of the SoftDevice controller, only the SoftDevice controller supports LLPM.

File: samples/bluetooth/peripheral_bms/README.rst [hunk 9]
```diff
@@ -19,14 +19,15 @@ It supports up to two simultaneous Client connections.
...
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns, nrf52840dk_nrf52840, nrf52dk_nrf52832
+   :rows: nrf5340pdk_nrf5340_cpuapp_and_cpuappns, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52dk_nrf52810
 
-* A Bluetooth Low Energy dongle or development kit
+Additionally, the sample has the following requirements:
 
+* A Bluetooth Low Energy dongle or development kit
```
Comment (joerchan) [inline samples/bluetooth/peripheral_bms/README.rst:30]: Isn't this covered by the above statement about dev-kits supported?

**Review by lemrey — COMMENTED** [review]
Found a typo.

**Review by joerchan — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Comment by umapraseeda** [comment]
@carlescufi ..This is ready to be merged.

**Review by carlescufi — APPROVED** [review]
